### ISSUE TITLE
Rethink the decision to enable e2ee by default in 1-1 chats

### ISSUE THREAD

**Issue by BrenBarn** [element-hq/element-meta#310]
## Description

This issue is really an Element meta-issue across all platforms.

The problem is that logging out of all devices results in any e2ee messages sent during that time becoming *permanently* unreadable, and this is not made clear to users, as evidenced by periodic questions from new users on the Matrix/Element rooms where people are surprised to find that logging out makes this kind of massive difference.  (See also vector-im/element-web#14820.)

There are several possible solutions:

### 1. Do not enable e2ee by default, ever.

I find this the most appealing option.  *Anything* that can *ever* make *any* messages *permanently* unreadable should not be enabled by default.  I feel that the decision to enable e2ee-by-default was made without adequate consideration of the magnitude of this failure mode.  Logging out is a *completely normal* thing to do; irrevocably losing messages is a complete code-red faceplant; that the one can lead directly to the other is a major problem.

### 2. Drastically increase the warnings before logging out and before turning on e2ee

Right now when you go to log out from Element you just get a confirmation dialog saying "Are you sure you want to sign out?"  It should be much scarier.  It should say something like "Warning!  Logging out of all devices may result in permanently losing access to messages sent while you are logged out.  Do you want to log out?"  Perhaps it could show a summary of currently logged in devices to help the user make the decision.

The point is that *anything that can lead to permanent message loss is a major decision* and should be signalled as such.  Either logging out is a fairly normal user action, or logging out can lead to permanent message loss; it can't be both.

A similar warning should be displayed before turning on e2ee in any room.  If e2ee remains the default for 1-1 chats, a similar message should be displayed before starting a 1-1 chat (e.g., "Warning!  Any messages in this chat may be lost if you log out of all devices.  If you don't want this, disable e2ee with this button...").

### 3. Consider some kind of intermediate level of e2ee that does not have these problems

I've not delved into the details of Matrix's encryption algorithms, so I don't really know what I'm talking about here.  But my understanding is that the can't-log-out problem arises because messages are encrypted with specific keys for each logged-in session, and these keys change frequently so a logged out device will "miss" the necessary key rotations.  An alternative would be to allow some kind of "placeholder" session, where when you log out on a particular device, it "freezes" an encryption key for all messages received until it logs in again, and stores the needed decryption key on the device, so when it logs in again the device itself has all the necessary information to decrypt messages left on the server in the interim.

This would provide a lower level of security, but for most people it would likely be enough.  (In fact, most people probably don't need e2ee at all, which is why I think I prefer option 1.)  Most people are probably fine with accepting some risk of unlikely snooping modes (e.g., their homeserver operator is malicious and tries to crack the stored messages) in exchange for being able to freely log in and out whenever they choose.

## Conclusion

The main point that I want to reiterate (even though I know I've reiterated it a number of times) is that **permanent message loss is a doomsday scenario for a messaging system**.  It is bad enough that it is even possible, but might be acceptable in cases where the user *explicitly* accepts the tradeoffs (e.g., for greater encryption secrecy).  But to enable it by default is just a bad idea; it makes it way too easy for people to lose their messages without ever even realizing that such a thing was possible.

Most people don't need e2ee to be on by default.  For most users, it is more important the intended recipient *does* see the message than that eavesdroppers *do not* see the message.  This is especially true because, for most users, the likelihood of anyone caring enough to eavesdrop on their messages is very low.

**Comment by BrenBarn** [comment]
Here are several other issues that all arise from the failure to get e2ee fully working before deploying it:

https://github.com/vector-im/element-web/issues/12275
https://github.com/vector-im/element-web/issues/9505
https://github.com/vector-im/element-web/issues/18304
https://github.com/vector-im/element-meta/issues/647
https://github.com/matrix-org/synapse/issues/2165

Why was e2ee enabled without first resolving *all* such issues, and why is it still default even though some of them are still open (in some cases after 3+ years)?

**Comment by BrenBarn** [comment]
Here's another: https://matrix.to/#/!OGEhHVWSdvArJzumhm:matrix.org/$mHg6rAqy842krayO15D9C8HNKdPc1fNh294DMJxboLI?via=matrix.org&via=libera.chat&via=wolfo.tech

**Comment by BrenBarn** [comment]
And again: https://matrix.to/#/!YTvKGNlinIzlkMTVRl:matrix.org/$n44NnpDuJmals1Bzn66z2nQhjFmb_fvG4k_LU3gDBkE?via=matrix.org&via=privacytools.io&via=mozilla.org

**Comment by t3chguy** [comment]
Instead of spamming everyone's mailbox each time you feel like posting an additional link, I suggest you make use of the edit feature.

**Comment by Zocker1999NET** [comment]
I would strongly disagree with disabling E2EE by default. Most users are not aware of the implications that comes with disabling encryption, especially in decentralized systems, where there is no central institution being a single attack vector with huge amounts of money and a reputation to lose, but we rely on homemade server administrators protecting our data.
When Element adds more warnings to E2EE chats to warn users about how easily they might lose their history, then there should also be warnings when creating unencrypted DM rooms: "Please be aware that messages & pictures in unencrypted rooms can be seen by Homeserver administrators. Consider these messages as public.".

But I agree that most Matrix clients make it too easy to create encrypted chats without having backed up your keys. And we should not rely on users thinking about that. So I propose the following (following "don't rely on the user" and "do not decrease privacy & security by default" and "don't warn users when they are safe"):

1. Remember the warning when first logging in into Element about backing up your security key in upper left corner? I think this warning should be bigger and prevent you from using Element until you either created the backup (which then allows you to log out from all devices without losing your history) or until you clicked huge warnings away which explain that you might lose your history & so on. And if you device upon not backing up your key, then Element should warn you when creating & participating in encrypted chat rooms. It should not warn you when you set up key backup good enough, otherwise we may loose people to WhatsApp & Signal (which nearly has the same problems as Element, but seems better at educating users to backup data & users and forcing them to not just log out) or, even worse, to platforms which don't bother to implement E2EE.

2. When logging in other devices, the same as above should happen until you verify your device (with the same possibility for expert users to click the warning away).

3. When logging out, Element should increase the warnings *if* it is the last (active) session which you log out (while considering sessions, which weren't online in the last few days as inactive).

4. Dehydrated devices should become default and be enabled for all (on update) as soon as it is ready to be out of experimental. Then users which repetitively log out their last session (e.g. because they use Element in the browser) can still read messages incoming while no session was signed in.

5. When dehydrated devices is disabled, users should also be warned when logging out the last active session that they may be unable to read messages incoming while being logged out & the warning should also have a button called "enable dehydrated devices & logout" to allow them easily to enable that as well.

**Comment by ShadowJonathan** [comment]
This, we shouldn't consider disabling E2EE by default, we should fix E2EE, and educate users.

I'd rather have Element take the stance that state/hostile surveillance is real, and protect its users, than ignore it's existence, and expose all users to it, in exchange for some minor UX improvement and user's ignorance in what they could do to protect themselves.

Encryption is not meant only for those who are actively aware of it, but also for those who don't, *especially* for those who don't.

**Comment by Fusseldieb** [comment]
Just pulling my hairs out with this design decision. 
I tried to implement Matrix/Element on a place I work at where the login should be as simple as possible. No keys or encryption, just a simple and effective communication tool where we can log in or out - and that's where the problems started. 
The client is forcing users to enable encryption, which is a major hassle, as we purposefully _DON'T_ wan't to enable it as people _WILL_ loose the keys and therefore messages, but it's being shoven down our throaths with NO option to disable, at least when it comes to DM's, which is just downright **_sad_**. It looks like an appealing platform, but the forced design choices are forcing me to slowly reconsider.
As much as I want to believe that E2EE with keys and whatnot will go right, _it won't_. There _WILL_ be lost stuff and there _will_ be a constant nag on my shoulder why certain stuff isn't readable anymore. I just want to disable it, even if needs to be done on a per-user basis on every client (since all devices are ours anyway).
I'm still looking for a way to disable it. If it doesn't work, I have to search for something else, unfortunately.

TLDR; Give the users the ability to turn the damn thing off, or get it from the .well-known/client config.

**Comment by t3chguy** [comment]
> TLDR; Give the users the ability to turn the damn thing off, or get it from the .well-known/client config.

You mean like it already does..? https://github.com/element-hq/element-meta/blob/develop/docs/client_well_known.md#encryption-configuration

**Comment by Fusseldieb** [comment]
> > TLDR; Give the users the ability to turn the damn thing off, or get it from the .well-known/client config.
> 
> You mean like it already does..? https://github.com/element-hq/element-meta/blob/develop/docs/client_well_known.md#encryption-configuration

I have already done that and set it to force_disable, and while it works for rooms, it **doesn't** for DM's. Element tries to start a DM as encrypted, and only fails because I told synapse to set it to an impossible level. 
Afaik we can only start a DM unencrypted if we create an unencrypted room and invite the person, which is, honestly, weird. Meh.
In other words, _no solution, only problem_.

---

EDIT: I kinda overcame this problem by using this custom module: https://github.com/digitalentity/matrix_encryption_disabler/
When I create a DM, it creates a room instead of a DM, but at least it doesn't error out anymore, and E2EE is disabled.

Still, give users an option to **disable E2EE** on the server as a whole. The constant key backup nag is annoying, especially since I don't even use any of it's features.

**Comment by richvdh** [comment]
> Rethink the decision to enable e2ee by default in 1-1 chats

We have no plans to change this.

**Comment by RokeJulianLockhart** [comment]
@richvdh, why?

**Comment by Fusseldieb** [comment]
> > Rethink the decision to enable e2ee by default in 1-1 chats
> 
> We have no plans to change this.

Don't change it, but give people **_the option_** to disable it on their servers.

**Comment by RokeJulianLockhart** [comment]
@Fusseldieb, I want to choose for my account. Perhaps, we should file issues solely about these?

**Comment by richvdh** [comment]
> [@richvdh](https://github.com/richvdh), why?

Because Element is marketed as being encrypted, and unlike Telegram and others we mean it.

> Don't change it, but give people **_the option_** to disable it on their servers.

Isn't that what https://github.com/element-hq/element-web/blob/develop/docs/e2ee.md#disabling-encryption-by-default does?
